### PR TITLE
Extracting is now slower based off your job

### PR DESCRIPTION
**Pull request by Kitsunemitsu**
## About The Pull Request
Extracting EGO as a role other than the EO now takes precious seconds.
Clerks get less of a punishment

## Why It's Good For The Game
I want it to be slower to extract yourself than to ask for help.

## Changelog
:cl:
balance: Extracting EGO as roles other than the EO is slowed down.
/:cl:

### CODE REVIEW

**Comment by Coxswain-Navigator** [comment]
This definitely warrants a test-merge.

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
Add a pop check. This is pointless at below ~10 pop.

**Comment by Kitsunemitsu** [comment]
Another excellent idea.

File: code/game/machinery/computer/abnormality_ego.dm [hunk 1]
```diff
@@ -56,13 +56,29 @@
...
+					extraction_delay = 50
+				if("Records Officer")	//No clue what the fuck you're doing here but go off.
+					extraction_delay = 30
+				if("Extraction Officer")
+					extraction_delay = 1
+			if(do_after(usr, extraction_delay, target = src))
+				var/obj/item/I = new E.item_path(get_turf(src))
+				A.stored_boxes -= E.cost
+				A.current_ego += I
+				to_chat(usr, "<span class='notice'>[I] has been dispensed!</span>")
```
Comment (Gboster-0) [inline code/game/machinery/computer/abnormality_ego.dm:74]: <span> my beloved
please change this to
to_chat(usr, span_notice("[I] has been dispensed!"))

Comment (Gboster-0) [reply code/game/machinery/computer/abnormality_ego.dm:74]: actually i'll probably revisit the file and change all the span's later. Nevermind this

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
Taking this off TM until requested changes are addressed.
If you have no time to update problematic PRs - remove them from TM until it is ready.

**Comment by Kitsunemitsu** [comment]
Fixed. It now requires 5 living players.

File: code/game/machinery/computer/abnormality_ego.dm [hunk 2]
```diff
@@ -56,13 +57,42 @@
...
+			var/extraction_delay = 300
+			var/living_players
+			switch(user_role)
+				if("Clerk")
+					extraction_delay = 50
+				if("Records Officer")	//No clue what the fuck you're doing here but go off.
+					extraction_delay = 30
+				if("Extraction Officer")
+					extraction_delay = 1
+			for(var/mob/living/carbon/human/L in GLOB.player_list)
```
Comment (EgorDinamit) [inline code/game/machinery/computer/abnormality_ego.dm:77]: Add a check for dead people or people off-z-level, such as ghost spawns(tutorial, for example).

File: code/game/machinery/computer/abnormality_ego.dm [hunk 3]
```diff
@@ -56,13 +57,42 @@
...
+				living_players++	//There's no fucking easier way to do this IIRC
+			if(living_players<5)	//you should have A clerk by 5 living players
```
Comment (EgorDinamit) [inline code/game/machinery/computer/abnormality_ego.dm:79]: 5 is too low in my opinion.

**Comment by Kitsunemitsu** [comment]
5 means that someone is bound to be a clerk or non-agent. There's often 1 agent 1 clerk.

File: code/game/machinery/computer/abnormality_ego.dm (hunk 3 above)
Comment (Kitsunemitsu) [reply code/game/machinery/computer/abnormality_ego.dm:79]: 5 means that someone is bound to be a clerk or non-agent. There's often 1 agent 1 clerk.

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
Set the minimum amount of players to 8 - 10. There is oftentimes a full agent roster, without clerks or EO (or sometimes, they just cryo).
We are not forcing people to play de-facto useless role below proper pop.

**Comment by Kitsunemitsu** [comment]
During testing, 5 is often enough for a single clerk to show up. Very rarely do 5 agents show up without a clerk showing up, and in that case, suck it up. You're trading away a support unit for a combat one.

**Comment by EgorDinamit** [comment]
Punishing people for playing the game the way it was intended will never be tolerated.
Increase the minimum to 8 or forget about this pointless anti-QOL PR.

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
I should really start to enforce it properly:
### Update the "Why is it good for the game" part. As of right now - you have 0 explanations on it, aside from why **you** want it.

File: code/game/machinery/computer/abnormality_ego.dm [hunk 4]
```diff
@@ -56,13 +57,42 @@
...
+			var/user_role = usr?.mind?.assigned_role
+			var/extraction_delay = 300
+			var/living_players
+			switch(user_role)
+				if("Clerk")
+					extraction_delay = 50
+				if("Records Officer")	//No clue what the fuck you're doing here but go off.
+					extraction_delay = 30
+				if("Extraction Officer")
+					extraction_delay = 1
```
Comment (EgorDinamit) [inline code/game/machinery/computer/abnormality_ego.dm:76]: 30 seconds is ridiculously long.
```suggestion
			var/extraction_delay = 10 SECONDS
			var/living_players
			switch(user_role)
				if("Clerk")
					extraction_delay = 3 SECONDS
				if("Records Officer")	//No clue what the fuck you're doing here but go off.
					extraction_delay = 2 SECONDS
				if("Extraction Officer")
					extraction_delay = 0
```
Add a check for if extraction delay is correct, so we don't have to run do_after for EO.

**Comment by EgorDinamit** [comment]
I suggest this PR is closed or reworked, as we have found a proper solution while discussing it on Discord.

**Comment by Kitsunemitsu** [comment]
The Anti-QOL Tag is actually not needed. 
1) We do not have a QOL Tag.
2) It gets lumped into the balance tag.

**Pull request closed without merging**